packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -99,124 +107,330 @@ const defaultProps = {
-      this.props.onKeyDown(event);
+    /* eslint-disable no-param-reassign */
+    if (DateUtil.isValidDate(formattedDate, dateFormat)) {
+      event.target.value = formattedDate;
+      onChange(event);
+    } else if (day === '' && month === '' && year === '') {
+      onChange(event);
+    } else if (day.length < 2 || month.length < 2 || year.length < 4) {
+      // Handles raw input change updates by combining inputs.
+      event.preventDefault();
+      if (localeFormat === 'MM-DD-YYYY') {
+        event.target.value = month + day + year;
+      } else if (localeFormat === 'DD-MM-YYYY') {
+        event.target.value = day + month + year;
+      } else {
+        event.target.value = year + month + day;
+      }
+      onChange(event);
     }
-  }
-
-  render() {
-    const {
-      buttonRefCallback,
-      inputAttributes,
-      intl,
-      isIncomplete,
-      isInvalid,
-      name,
-      onBlur,
-      onChange,
-      onClick,
-      onFocus,
-      onButtonFocus,
-      onKeyDown,
-      placeholder,
-      required,
-      value,
-      ...customProps
-    } = this.props;
-
-    this.onCalendarButtonClick = customProps.onCalendarButtonClick;
-    this.shouldShowPicker = customProps.shouldShowPicker;
-
-    delete customProps.onCalendarButtonClick;
-    delete customProps.shouldShowPicker;
-
-    const additionalInputProps = { ...customProps, ...inputAttributes };
-
-    const dateValue = DateUtil.convertToISO8601(value, DateUtil.getFormatByLocale(intl.locale));
-    const inputClasses = cx([
-      'input',
-      { 'is-invalid': isInvalid },
-      { 'is-incomplete': isIncomplete && required && !isInvalid },
-    ]);
-    const buttonClasses = cx([
-      'button',
-      { 'is-invalid': isInvalid },
-    ]);
-    const buttonText = intl.formatMessage({ id: 'Terra.datePicker.openCalendar' });
-
-    return (
-      <div className={cx('custom-input')}>
+    /* eslint-enable no-param-reassign */
+    setDate(inputValue, type);
+  };
+
+  const handleOnFocus = (event) => {
+    setFocused(true);
+    if (onFocus) {
+      onFocus(event);
+    }
+  };
+
+  const handleOnBlur = (event) => {
+    setFocused(false);
+    if (onBlur) {
+      onBlur(event);
+    }
+  };
+
+  const handleOnButtonClick = (event) => {
+    const attributes = { ...inputAttributes };
+    if (!attributes.readOnly && onCalendarButtonClick && onClick) {
+      onCalendarButtonClick(event, onClick);
+    }
+  };
+
+  const handleOnKeyDown = (event) => {
+    if (onKeyDown) {
+      onKeyDown(event);
+    }
+  };
+
+  const dateInputContainerClasses = cx([
+    'date-input-container',
+  ]);
+
+  const dateInputClasses = cx([
+    'date-input',
+    { 'is-focused': isFocused },
+    { 'is-invalid': isInvalid },
+    { 'is-incomplete': isIncomplete && required && !isInvalid },
+  ]);
+
+  const buttonClasses = cx([
+    'button',
+    { 'is-invalid': isInvalid },
+  ]);
+
+  const dateMonthInput = (
+    <Input
+      {...additionalInputProps}
```

Comment:
This is a bit tricky to pass the input props on each input. We aren't sure what these could tied to and usually these are accessibly attributes that are added for screen readers that are passed.